### ISSUE TITLE
[LLVM] stdc library function `localtime` causes exception due to O_NONBLOCK being unimplemented 

### ISSUE THREAD

**Issue by cagriuslu-adyen** [oracle/graal#4199]
**Describe GraalVM and your environment :**
```
GraalVM-21.3.0 and GraalVM-21.2.0
Enterprise Edition
JDK11
macOS Big Sur
amd64
Output of `java -Xinternalversion`: `Java HotSpot(TM) 64-Bit Server VM (11.0.12+8-LTS-jvmci-21.2-b06) for bsd-amd64 JRE (11.0.12+8-LTS-jvmci-21.2-b06), built on Jul  1 2021 04:44:57 by "graal2" with gcc 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.17)`
```

**Have you verified this issue still happens when using the latest snapshot?**
It happens on both 21.3.0 and 21.2.0

**Describe the issue**
Calling `localtime` from C code causes an exception with the message "O_NONBLOCK not implemented".
Diving into the musl-libc, `localtime` calls `tzset` which calls `__map_file` with O_NONBLOCK flag.
Issue can be circumvented by setting the TZ env var. `TZ=UTC` but we cannot do this in our production environment.

**How to compile the LLVM bitcode that causes the problem**
Bundled LLVM was used with no flags, as I've heard that bundled LLVM adds the necessary flags by default.

**Code snippet or code repository that reproduces the problem**
```
time_t now = time(NULL);
localtime(&now);
```

**Comment by rodrigar-mx** [comment]
Thanks for reporting this issue. We will take a look into it and get back to you.

**Comment by rodrigar-mx** [comment]
Hi @cagriuslu-adyen. I executed the following example and did not run into any error:
```
xxx@xxx-mac gs-3015 % java -version
openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment GraalVM CE 21.3.0 (build 11.0.13+7-jvmci-21.3-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 11.0.13+7-jvmci-21.3-b05, mixed mode, sharing)
xxx@xxx-mac gs-3015 % cat reproducer.c 
#include <stdio.h>
#include <time.h>
int main () {
 time_t rawtime = time(NULL); 
 struct tm *ltime;
 ltime = localtime( &rawtime );
 printf("Time: %s", asctime(ltime));
 return(0);
}
xxx@xxx-mac gs-3015 % export LLVM_TOOLCHAIN=$($JAVA_HOME/bin/lli --print-toolchain-path)
xxx@xxx-mac gs-3015 % $LLVM_TOOLCHAIN/clang reproducer.c -o reproducer 
xxx@xxx-mac gs-3015 % $JAVA_HOME/bin/lli reproducer
Time: Fri Jan 14 17:36:03 2022

```
Am I missing something on the reproducer? Otherwise, could you share more information on how you are compiling and executing the code.

**Comment by cagriuslu-adyen** [comment]
I should have made it clear that I'm using the LLVM toolchain bundled with 21.2.0, but run the compiled binary on 21.2.0 and 21.3.0. The reason I cannot use 21.3.0's bundled toolchain is this: https://github.com/oracle/graal/issues/4059

I've prepared a minimal example for you, in this example I've used 21.2.0. There is a precompiled binary in case you want to inspect it. `output.txt` is the output I see on my system.

[minimal-example.zip](https://github.com/oracle/graal/files/7880555/minimal-example.zip)

I don't think there is an unexpected bug here, but only an unimplemented feature. We have removed all O_NONBLOCK usages from our code. Any external library that uses O_NONBLOCK has became problem. We patched some external libraries, but the musl-libc we didn't change since it came precompiled with the bundled LLVM.

**Comment by cagriuslu-adyen** [comment]
If you haven't seen the issue does this mean O_NONBLOCK is already implemented with 21.3.0?

**Comment by rodrigar-mx** [comment]
Thanks for sharing @cagriuslu-adyen. I was not able to reproduce it because the error is only present when using the managed mode. I have now reported the issue internally. Any updates or questions, we will let you know.

**Comment by rschatz** [comment]
Should be fixed in the upcoming release (22.1).
